rumqttd/src/server/tls.rs

```diff
@@ -187,18 +187,10 @@ impl TLSAcceptor {
                 .collect();
 
             // Get private key
-            let key_file = File::open(key_path);
-            let key_file = key_file.map_err(|_| Error::ServerKeyNotFound(key_path.clone()))?;
-            let keys = rustls_pemfile::rsa_private_keys(&mut BufReader::new(key_file));
-            let keys = keys.map_err(|_| Error::InvalidServerKey(key_path.clone()))?;
-
-            // Get the first key
-            let key = match keys.first() {
-                Some(k) => k.clone(),
-                None => return Err(Error::InvalidServerKey(key_path.clone())),
-            };
-
-            (certs, PrivateKey(key))
+            let key = first_private_key_in_pemfile(key_path)
+                .map_err(|_| Error::InvalidServerKey(key_path.clone()))?;
```

Comment:
any type of `Error` returned from `first_private_key_in_pemfile` will get mapped to `InvalidServerKey`, that should not happen. 

maybe we can check `Error::ServerKeyNotFound` here and then pass in `key_file` to the fn.

Reply:
For this one, you want me to 

- Check if `Error::ServerKeyNotFound` -> if so, return it
- Otherwise, unwrap the existing error (without mapping to InvalidSeverKey)

Did I understand that right ?

Reply:
- yes
- ? is different from unwrap, if we unwrap, it would panic, we don't want that. ? will just return the error, which is what we intend

Reply:
pushed commit to fix this :+1: